### PR TITLE
adding what is the temporal server page, links, redirect

### PR DESCRIPTION
**Pull request by flossypurse**
## What does this PR do?

/doc/server-architecture -> /docs/content/what-is-the-temporal-server

+ links, redirects, new diagram(s)

Preview: https://deploy-preview-609--mystifying-fermi-1bc096.netlify.app/docs/content/what-is-a-temporal-cluster/

### CODE REVIEW

File: docs/content/what-is-the-temporal-server.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+readlist={[
+["What is Multi-Cluster Replication", "/docs/content/what-is-multi-cluster-replication"]
+]}
+/> -->
+
+<CenteredImage
+imagePath="/diagrams/temporal-system-entity-relationship.svg"
+imageSize="100"
+title="The Temporal Server topology"
+/>
```
Comment (sw-yx) [inline docs/content/what-is-the-temporal-server.md:28]: GREAT work on this!!!

File: docs/content/what-is-the-temporal-server.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+### History service
+
+The History Service manages Workflow state transitions.
+This service can scale internally by having multiple instances.
+
+It talks to the Frontend service, Matching service, and the database.
+
+It uses grpcPort 7234 to host the service handler.
+It uses port 6934 for membership related communication.
```
Comment (sw-yx) [inline docs/content/what-is-the-temporal-server.md:49]: for "what is" content i'm thinking this is a little low level - is there a better home for this info?

- eg "how to scale Temporal Server" or "how to debug Temporal Server" would need this info

File: docs/content/what-is-the-temporal-server.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+### Matching service
+
+The Matching Service is responsible for hosting Task Queues for Task dispatching.
+This service can scale internally by having multiple instances.
```
Comment (sw-yx) [inline docs/content/what-is-the-temporal-server.md:54]:
```suggestion
The Matching Service is responsible for hosting Task Queues for Task dispatching.
It is responsible for matching Workers to Tasks and routing new tasks to the appropriate queue.
This service can scale internally by having multiple instances.
```

File: docs/content/what-is-the-temporal-server.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+The Frontend service is responsible for all in-bound calls, including [Multi-cluster Replication](/docs/server/multi-cluster) related calls that are invoked by a remote cluster.
+
+It talks to the Matching service, History service, Worker service, and the database.
+
+It uses the grpcPort 7233 to host the service handler.
+It uses port 6933 for membership related communication with other hosts.
+
+### History service
+
+The History Service manages Workflow state transitions.
```
Comment (sw-yx) [inline docs/content/what-is-the-temporal-server.md:43]: more notes from Samar lecture

```suggestion
The History Service manages the internal state machine for Workflow execution.
When a new workflow task needs to be scheduled, History dispatches it to Matching.
```

File: docs/content/what-is-the-temporal-server.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+It talks to the Frontend service.
+
+It uses grpcPort 7239 to host the service handler.
+It uses port 6939 for membership related communication.
+
+### Database
+
+Cassandra, MySQL, and PostgreSQL schemas are supported and thus can be used as the Server's database.
+
+The database stores Tasks, Workflow data, Events, and visibility data.
```
Comment (sw-yx) [inline docs/content/what-is-the-temporal-server.md:74]:
```suggestion
The database stores:

 - **Tasks**: handling task dispatch
 - **Workflow state**:
	- **Execution** table: captures the mutable state tracked by Workflows
	- **History** table: an append only log of events
 - **Metadata**: for namespaces
 - **Visibility data**: driving operations like "show all open workflow executions". In production we recommend using ElasticSearch for high throughput usage.
```

File: docs/content/what-is-the-temporal-server.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+imageSize="100"
+title="The Temporal Server topology"
+/>
+
+### Frontend Service
+
+The Frontend service exposes a strongly typed [Proto API](https://github.com/temporalio/api/blob/master/temporal/api/workflowservice/v1/service.proto).
+
+The Frontend service is responsible for all in-bound calls, including [Multi-cluster Replication](/docs/server/multi-cluster) related calls that are invoked by a remote cluster.
+
```
Comment (sw-yx) [inline docs/content/what-is-the-temporal-server.md:35]:
```suggestion
The Frontend service is a singleton (not horizontally scalable) service responsible for rate limiting, authorizing, validating and routing all in-bound calls:
	- Domain CRUD
	- External Events
	- Poll APIs for workers
	- Visibility API requests
	- Admin CLI operations 
	- including [Multi-cluster Replication](/docs/server/multi-cluster) related calls that are invoked by a remote cluster.

Every inbound request has a workflow ID, which is hashed for routing by the Frontend service.
The Frontend service has visibility into the hash rings maintains membership info for routing, having knowledge of how many nodes are in the cluster.

Rate limiting is applied per host and per namespace.
```

todo: i think rate limiting and authorization info is pretty important - should we flesh this out?

File: docs/content/what-is-the-temporal-server.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+It uses the grpcPort 7233 to host the service handler.
+It uses port 6933 for membership related communication with other hosts.
+
+### History service
+
+The History Service manages Workflow state transitions.
+This service can scale internally by having multiple instances.
+
+It talks to the Frontend service, Matching service, and the database.
```
Comment (sw-yx) [inline docs/content/what-is-the-temporal-server.md:46]: more notes from Samar lecture

```suggestion
The History Service manages the internal state machine for Workflow execution.
When a new workflow task needs to be scheduled, History transactionally dispatches it to Matching via a transfer queue.


The History Service scales horizontally by individual shards, configured at cluster creation and static for the life of the cluster (so you should plan for scale and overprovision).
Each shard holds data (routing ID's, mutable state) and queues:
	- Transfer queue: Every time we do an update, we have to do a single transaction that updates the state of a workflow and creates a task to be dispatched
	- Timer queues: every time a Timer API is called, History creates a task to durably persist the timer
	- Replicator queue: only used for the experimental multi-cluster feature
```

**Review by sw-yx — APPROVED** [review]
added some comments to give more color and make it feel less superficial

File: docs/content/what-is-the-temporal-server.md (hunk 2 above)
Comment (flossypurse) [reply docs/content/what-is-the-temporal-server.md:49]: Totally - We can break this out. Not 100% sure what it looks like just yet, but probably part of the "everything I need to know about how to run Temporal" bucket.

**Pull request merged**